File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.py
class Solution:
    def firstStableIndex(self, nums: list[int], k: int) -> int:
        
        for i in range(len(nums)):
            maxLeft, minRight = -float("inf"), float("inf")
            for j in range(i + 1):
                maxLeft = max(maxLeft, nums[j])
            for j in range(i, len(nums)):
                minRight = min(minRight, nums[j])
            if maxLeft - minRight <= k:
                return i
        return -1

File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var firstStableIndex = function(nums, k) {
    
    for(var i = 0; i < nums.length; i++) {
        var maxLeft = -Infinity, minRight = Infinity
        for(let j = 0; j <= i; j++) {
            maxLeft = Math.max(maxLeft, nums[j])
        }
        for(let k = i; k < nums.length; k++) {
            minRight = Math.min(minRight, nums[k])
        }
        if (maxLeft - minRight <= k) {
            return i
        }
    }
    return -1
};